cmake/CatkinBuild.cmake

```diff
@@ -0,0 +1,72 @@
+# Set compiler flags
+set(CMAKE_CXX_STANDARD 14)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+  add_definitions(-O3)
+endif(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+
+# Find catkin macros and libraries
+find_package(catkin REQUIRED)
+find_package(Eigen3 REQUIRED)
+find_package(Boost REQUIRED COMPONENTS python)
+
+find_package(OpenMP REQUIRED)
+if (OpenMP_FOUND)
+  add_compile_options("${OpenMP_CXX_FLAGS}")
+  add_definitions(-DHAVE_OPENMP=${OpenMP_FOUND})
+endif()
+
+# Catkin package macro
+catkin_package(
+  INCLUDE_DIRS
+    nabo
+    ${CMAKE_SOURCE_DIR}
+    ${EIGEN3_INCLUDE_DIR}
+  LIBRARIES
+    nabo
+  DEPENDS
+    Boost
+)
+
+########################
+## Library definition ##
+########################
+# Nabo
+add_library(nabo
+  ${NABO_SRC}
+  ${NABO_HEADERS}
+)
+
+target_include_directories(nabo
+  PUBLIC
+    ${CMAKE_SOURCE_DIR}
+    ${CMAKE_SOURCE_DIR}/nabo
```

Comment:
You probably don't want these two, but instead `${CMAKE_CURRENT_SOURCE_DIR}`.

Reply:
I remember trying this in the past but it didn't work.

However, I do think that it makes sense to export the include directories

```
catkin_package(
  INCLUDE_DIRS
    ${CMAKE_SOURCE_DIR}
    ${EIGEN3_INCLUDE_DIR}
```

instead of 

```
catkin_package(
  INCLUDE_DIRS
    nabo
    ${CMAKE_SOURCE_DIR}
    ${EIGEN3_INCLUDE_DIR}
```

I just tried that and it was a cleaner solution than before.